runtime/jcl/common/proxy.c

```diff
@@ -23,6 +23,23 @@
 #include "jclglob.h"
 #include "jclprots.h"
 
+#if JAVA_SPEC_VERSION == 8
```

Comment:
Are these ifdefs needed?  Is this file used in later JDK releases or do they use Unsafe.defineClass instead? 

I guess my question is should this whole file only be included in JDK8?

Reply:
Looking into it, the later JDK releases don't use this file at all. JDK 11 seems to use `Unsafe.defineClass` to create the proxies, and JDK 12 goes through `ClassLoader.defineClassImpl`.

Should I remove those ifdefs then?